### DISCUSSION TITLE
Early return from templ component?

### DISCUSSION THREAD

**Discussion by ltdangle** [a-h/templ#934] [Q&A] [answered]
Hi all,
Was wondering if there an idiomatic way to implement an early return in the templ component. Something like:

```
templ MyComponent(err bool) {
if err {
  @DangerAlert()
  return 
}
... rest of the code ...
}

```
I think this is very useful when treating templates programmatically...

Thanks for suggestions!

**Comment by pharrisee** [comment, marked as answer]
a little late to the party but I've just tried this:

```golang
templ Index() {
	<h1 class="text-4xl font-semibold">Index</h1>
	<div>
		<h2 class="text-3xl mb-4">Buttons</h2>
	</div>
	if true {
		{{ return }}
	}
	<dialog id="the-dialog" class="rounded"></dialog>
}
```

And as would be hoped the `dialog` is not rendered.  

I'm still not sure that this would be something I'd expect to see in a real code base personally, since I'm not sure what side-effects there may be to returning early.  I can only imagine tag endings being missing but there's probably more side-effects that I'm not imagining.

This relies on the [Raw Go](https://templ.guide/syntax-and-usage/raw-go) experimental functionality.

**Comment by a-h** [comment]
I think of templ components as being like having an implicit yield of each node, so they don't fit into a very procedural style.

```templ
// Psuedo-code.
templ Component() {
  yield(<div>)
  yield("some text")
  yield("</div>")
}
```

I'd probably do what you're thinking about in Go code.

```go
func ErrorOrComponent(err error, c templ.Component) templ.Component {
  if err != nil {
    return DangerAlert()
  }
  return c
}
```

Then I could use it wherever I needed it:

```templ
templ MyComponent() {
  <div>Hello, World!</div>
}

templ IndexPage(err error) {
  @ErrorOrComponent(err, MyComponent())
}
```

**Comment by ltdangle** [comment]
Thank you all for input.

RawGo is what I'm looking for. I tried it and it compiles as advertised.
However, the tree sitter (in neovim, zed) does not parse code below the {{ Raw Go }} line, which is a deal-breaker. This and because  "this feature  may change or be removed in future" I opted for a workaround without using raw go.

A bummer, because I really want to to use plain go in the templates. 

Here is my vote to continue raw go development and make it a stable feature.

**Comment by a-h** [comment]
It's been promoted to stable in https://github.com/a-h/templ/commit/88eed0f46000ed0a0f93e9801bf20626708778ee, we just haven't cut a release yet.

**Reply by roborobs1023** [reply]
Has this been released yet?

**Reply by a-h** [reply]
Released right now, in https://github.com/a-h/templ/releases/tag/v0.2.793

**Reply by ltdangle** [reply]
Just confirming that as of v0.2.793 raw go works great. Lsp syntax highlighting also works as expected. Sweet!
